run-tests.pl

```diff
@@ -622,6 +622,13 @@ sub _push_test
        }
    }
 
+   if( exists $params{implementation_specific} ) {
+       my @tokens = split /::/, $SERVER_IMPL;
+       if ( $tokens[0] ne $params{implementation_specific} ) {
```

Comment:
This feels slightly too magical and fragile; could we give the Homeserver classes an `implementation_name` method (see `server_name` as an existing example) or something instead?

Reply:
I have added the `implementation_name` method to the `HomeserverFactory`, since that seems to be easily accessible from `run_tests.pl`.